src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/CsvFormatResponseIT.java

```diff
@@ -142,6 +156,460 @@ public void fieldOrderWithScriptFields() throws IOException {
+
+        List<String> lines = csvResult.getLines();
+        Assert.assertEquals(4, lines.size());
+        Assert.assertTrue("m,36,31.0,647425.0",
+                lines.contains("m,36,31.0,647425.0") || lines.contains("m,36,647425.0,31.0"));
+        Assert.assertTrue("m,35,28.0,678337.0",
+                lines.contains("m,35,28.0,678337.0") || lines.contains("m,35,678337.0,28.0"));
+        Assert.assertTrue("f,36,21.0,505660.0",
+                lines.contains("f,36,21.0,505660.0") || lines.contains("f,36,505660.0,21.0"));
+        Assert.assertTrue("f,35,24.0,472771.0",
+                lines.contains("f,35,24.0,472771.0") || lines.contains("f,35,472771.0,24.0"));
+
+    }
+
+    @Test
+    public void dateHistogramTest() throws Exception {
+        String query = String.format(Locale.ROOT, "select count(*) from %s/online" +
+                " group by date_histogram('field'='insert_time','interval'='4d','alias'='days')", TEST_INDEX_ONLINE);
+        CSVResult csvResult = executeCsvRequest(query, false);
+        List<String> headers = csvResult.getHeaders();
+        Assert.assertEquals(2, headers.size());
+        Assert.assertEquals("days", headers.get(0));
+        Assert.assertEquals("COUNT(*)", headers.get(1));
+
+        List<String> lines = csvResult.getLines();
+        Assert.assertEquals(3, lines.size());
+        Assert.assertTrue("2014-08-14 00:00:00,477.0", lines.contains("2014-08-14 00:00:00,477.0"));
+        Assert.assertTrue("2014-08-18 00:00:00,5664.0", lines.contains("2014-08-18 00:00:00,5664.0"));
+        Assert.assertTrue("2014-08-22 00:00:00,3795.0", lines.contains("2014-08-22 00:00:00,3795.0"));
+
+    }
+
+    @Test
+    public void statsAggregationTest() throws Exception {
+        String query = String.format(Locale.ROOT, "SELECT STATS(age) FROM %s/account", TEST_INDEX_ACCOUNT);
+        CSVResult csvResult = executeCsvRequest(query, false);
+        List<String> headers = csvResult.getHeaders();
+        Assert.assertEquals(5, headers.size());
+        Assert.assertEquals("STATS(age).count", headers.get(0));
+        Assert.assertEquals("STATS(age).sum", headers.get(1));
+        Assert.assertEquals("STATS(age).avg", headers.get(2));
+        Assert.assertEquals("STATS(age).min", headers.get(3));
+        Assert.assertEquals("STATS(age).max", headers.get(4));
+
+        List<String> lines = csvResult.getLines();
+        Assert.assertEquals(1, lines.size());
+        Assert.assertEquals("1000,30171.0,30.171,20.0,40.0", lines.get(0));
+
+    }
+
+    @Test
+    public void extendedStatsAggregationTest() throws Exception {
+        String query = String.format(Locale.ROOT, "SELECT EXTENDED_STATS(age) FROM %s/account", TEST_INDEX_ACCOUNT);
+        CSVResult csvResult = executeCsvRequest(query, false);
+        List<String> headers = csvResult.getHeaders();
+        Assert.assertEquals(8, headers.size());
+        Assert.assertEquals("EXTENDED_STATS(age).count", headers.get(0));
```

Comment:
Assert equals between 2 Lists is a little cleaner.

Reply:
makes sense, updated